### DISCUSSION TITLE
Connection Status Widget

### DISCUSSION THREAD

**Discussion by ConfidentStorm** [mullvad/mullvadvpn-app#9062] [Feature requests & Ideas]
## Issue report

Operating system: 
iOS 15.1
App version: 
2021.3
### Issue description

Hello,

I would love for you all to introduce a widget for iOS. I would prefer this widget to simply be a “Connection Status” widget. After Apple introduced the notch on the iPhones, they hid the VPN Connection Status Icon in the Control Center. I cannot keep Mullvad connected all the time on my iPhone because some streaming apps will not work for me while Mullvad is connected. I will often forget to reconnect Mullvad when I am done streaming content. Due to this, I often try to check if I am connected by pulling down Control Center before I launch an app whose network activity I would prefer to be encrypted by Mullvad.  However, I often pull down the notification center instead and then get frustrated just trying to see if I am connected. This might not be an issue for users that own smaller iPhones but with the Max it is a common occurrence. 
I know you all have finite amount of time and resources, so I understand this is not a priority. The VPN I used before transitioning to Mullvad was Tunnel Bear. They have a great “Connection Status Widget” that I would often quickly swipe over to check and then go about my business (I have attached 
 below two screen recordings to demonstrate how theirs functions). I know that this is a trivial request, I could simply launch the Mullvad app or check the settings app to determine my connection status. But I really love the way I have my home screen set up and being able to quickly swipe anywhere on the screen and check the widgets page before launching an app would be perfect for me. I love your MacOS Menu Bar Green/Red Lock Icon, I use it all the time to check my Mac’s connection status, maybe you all could implement a version of it for the iOS widget. 

Thank you for all you do,
Confident Storm

Disconnected:
https://user-images.githubusercontent.com/93688580/140610679-d512c60b-3edd-437b-bb67-8dd4417ba9da.MP4

Connected:

https://user-images.githubusercontent.com/93688580/140610698-b69270a2-c780-4aee-a61a-d75d83272c28.mp4

**Comment by tort-oise** [comment]
This isn't entirely unreasonable. The proof of concept is there.
Sounds like a great addition!

**Comment by ConfidentStorm** [comment]
Great to hear, I look forward to it! Thank you!!

**Comment by ScriptInfra** [comment]
Has there been any movement on creating a widget for iOS? This would be an incredible quality of life update/feature. I often am forced to disconnect from the VPN to cast shows on Netflix and it's easy to forget to reconnect since iOS does not show the connection status as described above.

**Comment by pronebird** [comment]
Noted.

**Comment by yuriyhanysh** [comment]
@pronebird, are you open to contributions? I want to implement this one if you have not started working on it.

**Comment by pronebird** [comment]
@yuriyhanysh Yes, we accept contributions.

**Comment by AstudilloMarillo** [comment]
Any update on this in the meantime? Seems like there was a positive response to this feature request and willingness to implement it. Is there any progress?

**Comment by pronebird** [comment]
> Any update on this in the meantime? Seems like there was a positive response to this feature request and willingness to implement it. Is there any progress?

I think we looked at implementing widget ourselves recently and found out that basically there would be no way for us to provide real-time updates to widget because of the whole "budgeting" system on iOS where widgets receive limited amount of updates per day. 

In the past so-called "Today widget"s used to be more flexible if I remember that right, but those were deprecated in favor of new widget kit which is all about budgeting and man oh man not in this economy...

**Comment by Duckbilled** [comment]
May I drop a suggestion (and fingers crossed I haven't skimmed to roughly over earlier comments), I used IVPN for a while and I like their simple implementation as can be seen from their app store screenshot [(last one)](https://apps.apple.com/us/app/ivpn-secure-vpn-for-privacy/id1193122683).

The only thing I could imagine it to be more fancy, is if it allowed for the option to select a few favourites.

**Comment by metametapod** [comment]
A simple implementation like IVPN's would be useful. Probably more involved but I would even use a widget on macOS e.g. when connected via the `mullvad` cli daemon and not running the menubar app.

**Comment by robolth** [comment]
I would like to +1 this feature request, I too often forget to turn the VPN back on

**Comment by Dev98248** [comment]
+1 to this as well. It's really hard to notice if my VPN is on or off on ios

**Comment by rak00001** [comment]
+1 for sure, just such a good qol enhancement.
